jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixGroovyScriptParametersProvider.java

```diff
@@ -18,60 +18,100 @@
+                                              TimeSource timeSource,
+                                              ApplicationContext applicationContext,
+                                              Messages messages) {
+        this.reportsProperties = reportsProperties;
+        this.metadata = metadata;
+        this.dataManager = dataManager;
+        this.currentAuthentication = currentAuthentication;
+        this.timeSource = timeSource;
+        this.applicationContext = applicationContext;
+        this.messages = messages;
+    }
 
     @Override
-    public Map<String, Object> prepareParameters(ReportQuery reportQuery, BandData parentBand, Map<String, Object> reportParameters) {
+    public Map<String, Object> getParametersForDatasetParameters(ReportQuery reportQuery, BandData parentBand, Map<String, Object> reportParameters) {
+        Map<String, Object> scriptParams = getCommonParameters();
 
-        Map<String, Object> scriptParams = new HashMap<>();
         scriptParams.put("reportQuery", reportQuery);
         scriptParams.put("parentBand", parentBand);
         scriptParams.put("params", reportParameters);
-        scriptParams.put("currentAuthentication", currentAuthentication);
-        scriptParams.put("metadata", metadata);
-        scriptParams.put("dataManager", dataManager);
+        scriptParams.put("showErrorMessage", throwDatasetValidationException());
         scriptParams.put("timeSource", timeSource);
-        scriptParams.put("applicationContext", applicationContext);
-        //todo replace MethodClosure
-//        scriptParams.put("validationException", new MethodClosure(this, "validationException"));
 
         return scriptParams;
     }
 
-    protected void validationException(String message) {
-        throw new ValidationException(message);
+    @Override
+    public Map<String, Object> getParametersForValidationParameters() {
+        Map<String, Object> params = getCommonParameters();
+        params.put("showErrorMessage", throwParameterValidationException());
+        return params;
+    }
+
+    protected Map<String, Object> getCommonParameters() {
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("currentAuthentication", currentAuthentication);
+        params.put("applicationContext", applicationContext);
+        params.put("dataManager", dataManager);
+        params.put("metadata", metadata);
+
+        return params;
+    }
+
+    protected ExceptionCallable throwParameterValidationException() {
+        return (arguments) -> {
+            String message = (String) Arrays.stream(arguments)
+                    .findFirst()
+                    .orElse(messages.getMessage("validationFieldFail.defaultDescriptionMessage"));
+            throw new ReportParametersValidationException(message);
+        };
+    }
+
+    protected ExceptionCallable throwDatasetValidationException() {
```

Comment:
Maybe we should change the method name? When I see the method called throwAaaException I expect that the method will throw an exception. Your method creates the ExceptionCallable object, right? So maybe name the method "createDatasetValidationExceptionCallable" or something like this?